src/message/poll.rs

```diff
@@ -29,41 +30,24 @@ impl Default for MessagePoll<'static> {
 
 /// Type that represents the &lt;trnData&gt; tag for message poll response
 #[derive(Deserialize, Debug)]
-pub struct MessageDomainTransferData {
-    /// The name of the domain under transfer
-    #[serde(rename = "domain:name", alias = "name")]
-    pub name: StringValue<'static>,
-    /// The domain transfer status
-    #[serde(rename = "domain:trStatus", alias = "trStatus")]
-    pub transfer_status: StringValue<'static>,
-    /// The epp user who requested the transfer
-    #[serde(rename = "domain:reID", alias = "reID")]
-    pub requester_id: StringValue<'static>,
-    /// The date of the transfer request
-    #[serde(rename = "domain:reDate", alias = "reDate")]
-    pub requested_at: StringValue<'static>,
-    /// The epp user who should acknowledge the transfer request
-    #[serde(rename = "domain:acID", alias = "acID")]
-    pub ack_id: StringValue<'static>,
-    /// The date by which the transfer request should be acknowledged
-    #[serde(rename = "domain:acDate", alias = "acDate")]
-    pub ack_by: StringValue<'static>,
-    /// The domain expiry date
-    #[serde(rename = "domain:exDate", alias = "exDate")]
-    pub expiring_at: Option<StringValue<'static>>,
+pub enum MessageData {
+    /// Data under the &lt;domain:trnData&gt; tag
+    #[serde(rename = "domain:trnData")]
+    DomainTransferResponseData(DomainTransferResponseData),
```

Comment:
Nice deduplication!

Maybe ditch the `ResponseData` suffix of the variant name here?

Reply:
Great idea